book/listings/interface_builder/2/window/imp.rs

```diff
@@ -0,0 +1,58 @@
+use glib::subclass::InitializingObject;
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{Button, CompositeTemplate};
+
+// ANCHOR: object
+// Object holding the state
+#[derive(CompositeTemplate, Default)]
+#[template(file = "window.ui")]
+pub struct Window {
+    #[template_child]
+    pub button: TemplateChild<Button>,
+}
+// ANCHOR_END: object
+
+// ANCHOR: subclass
+// The central trait for subclassing a GObject
+#[glib::object_subclass]
+impl ObjectSubclass for Window {
+    const NAME: &'static str = "MyGtkAppWindow";
+    type Type = super::Window;
+    type ParentType = gtk::ApplicationWindow;
+
+    fn class_init(klass: &mut Self::Class) {
+        Self::bind_template(klass);
+    }
+
+    fn instance_init(obj: &InitializingObject<Self>) {
+        obj.init_template();
+    }
+}
+// ANCHOR_END: subclass
+
+// ANCHOR: object_impl
+// Trait shared by all GObjects
+impl ObjectImpl for Window {
+    fn constructed(&self, obj: &Self::Type) {
+        // Call "constructed" on parent
+        self.parent_constructed(obj);
```

Comment:
chain up at the end i suppose?

Reply:
I wasn't sure if it belongs at the beginning or end. I'll move it if that's the correct position.

Reply:
From what I have seen from libadwaita and gtk it is done at the start. But there might be a niche reason to do it at the end.

In most places in Rust I see it at the end, and it seems to be done that way just because it is the way it has been done.

Reply:
Who does it at the end? All my code in gst-plugins-rs does it at the beginning too and that's more correct in most cases

Reply:
Also the gtk-rs/glib tests and docs. Doing it at the end has the potential of failing, depending on the parent class.

Reply:
> Who does it at the end? 

Apps.

Reply:
That sounds like a bad idea unless they have a special reason for that. For other virtual methods that might make sense but not for constructed. It's always a decision for each virtual method when you want to have your behaviour and when the base behaviour.

Reply:
Someone fix those then please :)